### PR TITLE
flamegraph collector v1

### PR DESCRIPTION
**Pull request by Ryado**
This is the initial version of the flamegraph collector, Collects a flamegraph representation of trace logs using the `system.trace_log` table. Currently collects a non-collapsed version in a single file format.

Please check if the MIT license header is ok

### CODE REVIEW

**Comment by Ryado** [comment]
working on fixing the tests

File: internal/collectors/clickhouse/flamegraph.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		return &data.DiagnosticBundle{}, err
+	}
+
+	frames := make(map[string]data.Frame)
+
+	var frameErrors []error
+
+	//Trace Query Template
+	var (
+		traceSQLTemplate = `
```
Comment (gingerwizard) [inline internal/collectors/clickhouse/flamegraph.go:83]: id expect to see this declared as const outside the method..just makes the method more readable as its not essential to the logic

File: internal/collectors/clickhouse/flamegraph.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+					) AS stack
+				FROM {from}
+				WHERE {where}
+				GROUP BY host_name, query_id, trace_type, trace
+				LIMIT {rowLimit}
+				SETTINGS allow_introspection_functions=1
+				`
+	)
+
+	stackWhere := " trace_type IN ('" + strings.Join(traceTypes, "','") + "')"
```
Comment (gingerwizard) [inline internal/collectors/clickhouse/flamegraph.go:106]: can we use fmt.Sprintf for code consistency

File: internal/collectors/clickhouse/flamegraph.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+func (sc *FlamegraphCollector) Configuration() config.Configuration {
+	return config.Configuration{
+		Params: []config.ConfigParam{
+			config.IntParam{
+				Value: 100000,
+				Param: config.NewParam("row_limit", "Limit rows on supported queries.", false),
+			},
+			config.StringListParam{
+				Values: []string{"Real", "CPU", "Memory", "MemorySample"},
```
Comment (gingerwizard) [inline internal/collectors/clickhouse/flamegraph.go:166]: Would we not want a different frame per trace type?

File: internal/collectors/clickhouse/flamegraph.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+	for k, v := range context {
+		args[i] = "{" + k + "}"
+		args[i+1] = fmt.Sprint(v)
+		i += 2
+	}
+	return strings.NewReplacer(args...).Replace(sqlTemplate)
+}
+
+func (sc *FlamegraphCollector) Configuration() config.Configuration {
+	return config.Configuration{
```
Comment (gingerwizard) [inline internal/collectors/clickhouse/flamegraph.go:159]: did we remove the ability to filter byu query id? Do we not see a need for it?

**Review by gingerwizard — COMMENTED** [review]
Initial review

**Comment by gingerwizard** [comment]
@Ryado you have a few conflicts and left a few comments.

**Comment by Ryado** [comment]
I'll close this PR and re-open another once it's merged with the main repo

**Pull request closed without merging**